src/providers/info.rs

```diff
@@ -102,19 +102,27 @@ pub trait InfoProvider {
 }
 
 pub struct DefaultInfoProvider<'p, C: Cache> {
+    pub chain_id: &'p str,
```

Comment:
@rmeissner I removed the mutex cache, as `chain_id` is now a "constructor" parameter